tenant/store.go

```diff
@@ -38,10 +40,11 @@ func (tenantStore *tenantStore) Entities() []interface{} {
 }
 
 type Tenant struct {
-	Id       uint64 `sql:"AUTO_INCREMENT"`
-	Name     string
-	Segments []Segment
-	Seq      uint64
+	ID         uint64 `sql:"AUTO_INCREMENT"`
+	ExternalID string `sql:"not null;unique"`
```

Comment:
This is a very good question (and it's understood that this is not about these lines of code here, but more a general topic).

I would say that Romana components should use Romana IDs wherever possible. However, just like with a database, where you can lookup on the primary key, but also can construct queries on the other fields, I wonder if it might be useful to provide a simple query interface for arbitrary (or selected) fields. So, while normally we would GET an entry like this:

    bar.com/elements/123456

Maybe we can also offer something like:

    bar.com/elements/?field=abcdefg

That way, a component working mostly with external entities (an adaptor component, for example) can easily perform certain queries without having to perform a separate lookup.

I don't see a case for treating external IDs in any particularly special way. For our Romana components they are just another field in a data structure. The writing and interpretation of external IDs should be limited to only the adapter components.